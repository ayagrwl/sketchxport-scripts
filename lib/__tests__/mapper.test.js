const { mapColors, mapTextStyles, formatColor } = require('../mappers');
const sampleOutput = require('../../__mocks__/sample_dump.json');

const [ULTRAMARINE, ...COLORS] = sampleOutput.document.assets.colors;
const TEXT_STYLES = sampleOutput.document.layerTextStyles;

describe('[mapper.js] test suite', () => {
  describe('-> #formatColor:', () => {
    it('given RGB NSColor, should transform it to a nicely parsed object', () => {
      expect(formatColor(ULTRAMARINE)).toMatchSnapshot();
    });
  });
  describe('-> #mapColors:', () => {
    it('given the document colors, should transform all colors nicely', () => {
      expect(mapColors(COLORS)).toMatchSnapshot();
    });
  });
  describe('-> #mapTextStyles:', () => {
    it('given the text styles, should transform them nicely', () => {
      const parsed = mapTextStyles(TEXT_STYLES);
      expect(parsed).toMatchSnapshot();

      const [heading1, heading2, paragraph] = parsed;
      expect(heading1.id).toBe('Heading 1');
      expect(heading2.id).toBe('Heading 2');
      expect(paragraph.id).toBe('Paragraph');
    });
  });
});
