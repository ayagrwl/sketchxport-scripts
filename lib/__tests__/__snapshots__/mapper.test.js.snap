// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[mapper.js] test suite -> #formatColor: given RGB NSColor, should transform it to a nicely parsed object 1`] = `
Object {
  "alpha": 1,
  "blue": "159",
  "green": "6",
  "hex": "10069f",
  "hsl": Array [
    244,
    93,
    32,
  ],
  "name": "ultramarine",
  "red": "16",
  "rgb": "16, 6, 159",
  "rgba": "16, 6, 159, 1",
}
`;

exports[`[mapper.js] test suite -> #mapColors: given the document colors, should transform all colors nicely 1`] = `
Array [
  Object {
    "alpha": 1,
    "blue": "80",
    "green": "0",
    "hex": "e50050",
    "hsl": Array [
      339,
      100,
      45,
    ],
    "name": "amaranth",
    "red": "229",
    "rgb": "229, 0, 80",
    "rgba": "229, 0, 80, 1",
  },
  Object {
    "alpha": 1,
    "blue": "78",
    "green": "190",
    "hex": "2cbe4e",
    "hsl": Array [
      134,
      62,
      46,
    ],
    "name": "apple",
    "red": "44",
    "rgb": "44, 190, 78",
    "rgba": "44, 190, 78, 1",
  },
  Object {
    "alpha": 1,
    "blue": "0",
    "green": "0",
    "hex": "000000",
    "hsl": Array [
      0,
      0,
      0,
    ],
    "name": "black",
    "red": "0",
    "rgb": "0, 0, 0",
    "rgba": "0, 0, 0, 1",
  },
]
`;

exports[`[mapper.js] test suite -> #mapTextStyles: given the text styles, should transform them nicely 1`] = `
Array [
  Object {
    "borderBottom": null,
    "color": Object {
      "alpha": 1,
      "blue": "159",
      "green": "6",
      "hex": "10069f",
      "hsl": Array [
        244,
        93,
        32,
      ],
      "name": "ultramarine",
      "red": "16",
      "rgb": "16, 6, 159",
      "rgba": "16, 6, 159, 1",
    },
    "family": "FiraSans-Regular",
    "id": "Heading 1",
    "kerning": null,
    "size": 48,
    "weight": null,
  },
  Object {
    "borderBottom": null,
    "color": Object {
      "alpha": 1,
      "blue": "159",
      "green": "6",
      "hex": "10069f",
      "hsl": Array [
        244,
        93,
        32,
      ],
      "name": "ultramarine",
      "red": "16",
      "rgb": "16, 6, 159",
      "rgba": "16, 6, 159, 1",
    },
    "family": "FiraSans-Regular",
    "id": "Heading 2",
    "kerning": null,
    "size": 36,
    "weight": null,
  },
  Object {
    "borderBottom": null,
    "color": Object {
      "alpha": 1,
      "blue": "0",
      "green": "0",
      "hex": "000000",
      "hsl": Array [
        0,
        0,
        0,
      ],
      "name": "black",
      "red": "0",
      "rgb": "0, 0, 0",
      "rgba": "0, 0, 0, 1",
    },
    "family": "FiraSans-Regular",
    "id": "Paragraph",
    "kerning": null,
    "size": 18,
    "weight": null,
  },
]
`;
